protocols/shell-geometry-tracker-v1.xml

```diff
@@ -0,0 +1,260 @@
+        whenever insets change. This event is double-buffered
+        and must be followed by a
+        'shell_foreign_toplevel_geometry_tracker_manager_v1.done' event.
+      </description>
+
+      <arg name="left" type="int"/>
+      <arg name="top" type="int"/>
+      <arg name="right" type="int"/>
+      <arg name="bottom" type="int"/>
+    </event>
+
+    <event name="entered">
+      <description name="an entity has entered the tracker area">
+        This event reports that a toplevel has entered the tracking
+        area. Untracked entities must not receive 'entered' events.
+      </description>
+
+      <arg name="member" type="new_id" interface="shell_foreign_toplevel_geometry_tracker_member_v1"/>
+    </evnt>
+
+    <request name="track_toplevels">
+      <description summary="begins tracking toplevels">
+        This request asks the compositor to track all toplevels created by the
+        given toplevel list that enter the tracking area.
+
+        Upon request, all toplevels that qualify as members of the tracker
+        must be provided via an 'enter' event.
+
+        Attempting to set the toplevel list more than once is an error, as
+        is destroying the toplevel list prior to the destruction of
+        this object.
+      </description>
+
+      <arg name="list" type="object" interface="ext_foreign_toplevel_list_v1"/>
+    </request>
+  </interface>
+
+  <interface name="shell_foreign_toplevel_geometry_tracker_member_v1" version="1">
+    <description summary="tracked entity visible on a geometry tracker">
+      A toplevel tracked by a geometry tracker.
+
+      Upon creation, toplevel, geometry, and optionally stacking order
+      events must be sent by the compositor.
+    </description>
+
+    <request name="destroy" type="destructor">
+      <description summary="destroy the member">
+        Destroys the tracker member object.
+      </description>
+    </request>
+
+    <event name="left">
+      <description summary="the toplevel has left the tracked area">
+        The toplevel has left the tracked area. A toplevel must not be
+        considered to have left until the toplevel's geometry has no
+        intersection with the geometry tracker.
+
+        Following this event, the member object should be destroyed and no more
+        events will be sent to it. If the toplevel later re-enters the tracked
+        area it will do so as a new object.
+
+        This event is double-buffered and must be followed by a
+        'shell_foreign_toplevel_geometry_tracker_manager_v1.done' event.
+      </description>
+    </event>
+
+    <event name="geometry">
+      <description summary="geometry of the toplevel has changed">
+        New toplevel geometry information is available.
+        This event is sent upon creation of the member object or changes
+        to the toplevel's geometry.
+
+        Reported geometry should include window decorations such as borders
+        but not shadows.
```

Comment:
leave this compositor policy what decos to include, but suggest excluding "non-window" extents like shadows